File: src/Deviser.Core/Deviser.Admin.Web/ClientApp/src/app/common/domain-types/model-config.ts
import { KeyField } from "./key-field";
import { GridConfig } from "./grid-config";
import { FormConfig } from './form-config';
import { CustomForm } from './custom-form';
import { TreeConfig } from './tree-config';

export interface ModelConfig {
    customForms: { [key: string]: CustomForm }
    formConfig: FormConfig;
    gridConfig: GridConfig;
    keyField: KeyField;
    treeConfig: TreeConfig;
}

File: src/Deviser.Core/Deviser.Admin/Config/ModelConfig.cs
using System.Collections.Generic;
using Deviser.Admin.Config;

namespace Deviser.Admin
{
    public class ModelConfig : IModelConfig
    {
        public IDictionary<string, CustomForm> CustomForms { get; }
        public IFormConfig FormConfig { get; }
        public IGridConfig GridConfig { get; }
        public KeyField KeyField { get; }
        public ITreeConfig TreeConfig { get; }
        public ModelConfig()
        {
            CustomForms = new Dictionary<string, CustomForm>();
            FormConfig = new FormConfig();
            GridConfig = new GridConfig();
            KeyField = new KeyField();
            TreeConfig = new TreeConfig();
        }
    }
}